AC3-2-Final-Retake-Fireblog/AC3-2-Final-Retake-Fireblog/FeedViewController.swift

```diff
@@ -0,0 +1,220 @@
+                let timestamp = value["timestamp"] as? Double,
+                let type = value["type"] as? String,
+                let text = value["text"] as? String? {
+                
+                
+                switch type {
+                case type where type == PostType.text.rawValue:
+                    let textPost = Post(email: email, type: type, timestamp: timestamp, postID: key, text: text, image: nil)
+                    completion(textPost)
+                case type where type == PostType.image.rawValue:
+                    self.getPostImageFromStorage(key) { (image) in
+                        let imagePost = Post(email: email, type: type, timestamp: timestamp, postID: key, text: nil, image: image)
+                        completion(imagePost)
+                    }
+                default:
+                    continue
+                }
+            }
+        }
+    }
+    
+    func getPostImageFromStorage(_ key: String, completion: @escaping (UIImage) -> ()) {
+        
+        storageReference = FIRStorage.storage().reference(forURL: "gs://ac-32-final-retake.appspot.com/").child("images/").child("\(key)")
+        
+        storageReference.data(withMaxSize: 1 * 1024 * 1024) { (data, error) in
+            if error != nil {
+                print("ERROR DOWNLOADING IMAGE: \(error!.localizedDescription)")
+            }
+            if let postImage = UIImage(data: data!) {
+                completion(postImage)
+            }
+        }.resume()
+    }
+    
+    
+    // MARK: Lazy Instantiation
+    
+    lazy var tableView: UITableView = {
+        let tv = UITableView()
+        tv.translatesAutoresizingMaskIntoConstraints = false
+        return tv
+    }()
+    
+    // MARK: TableView Data Source Methods
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        let postsSortedByDate = self.posts.sorted() { $0.timestamp > $1.timestamp}
+        return postsSortedByDate.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "FeedCell", for: indexPath) as! FeedTableViewCell
+        let postsSortedByDate = self.posts.sorted() { $0.timestamp > $1.timestamp}
+        
+        let post = postsSortedByDate[indexPath.row]
+        
+        cell.userEmailLabel.text = post.madeByUserWithEmail
+        
+        let timeInterval = TimeInterval(post.timestamp)
+        let dateFromTimeInterval = NSDate(timeIntervalSinceReferenceDate: timeInterval)
+        
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "M/d/yy, h:mm a"
+        
+        let postDateString = dateFormatter.string(from: dateFromTimeInterval as Date)
+        cell.timestampLabel.text = postDateString
+        
+        if post.type == PostType.text.rawValue {
+            cell.postImageView.removeFromSuperview()
```

Comment:
I really like this strategy of removing views from the contentView, but the cell holds on to them in memory and can add them back in.